### PR TITLE
BridgeMessageId=>TokenMessageId

### PR DESCRIPTION
**Pull request by xiaoch05**
To avoid the type conflict of this update.
https://github.com/paritytech/parity-bridges-common/pull/1152

### CODE REVIEW

**Comment by xiaoch05** [comment]
@HackFisher  the docs is here https://github.com/darwinia-network/darwinia-common/blob/master/frame/support/src/lib.rs#L103-L107.
and for sub<>sub bridge, it combines laneid and nonce. We leave 4 bytes to indict chainid in the future.

and this message_id also used in mapping-token-factory at https://github.com/darwinia-network/darwinia-bridge-sol/blob/master/contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol#L154

**Comment by xiaoch05** [comment]
@HackFisher  the docs is here https://github.com/darwinia-network/darwinia-common/blob/master/frame/support/src/lib.rs#L103-L107.
and for sub<>sub bridge, it combines laneid and nonce. We leave 4 bytes to indict chainid in the future.

and this message_id also used in mapping-token-factory at https://github.com/darwinia-network/darwinia-bridge-sol/blob/master/contracts/mapping-token/contracts/darwinia/DarwiniaMappingTokenFactory.sol#L154

**Comment by HackFisher** [comment]
Since this is only used in wormhole and token protocol, please rename it to TokenMessageId, and also related traits & methods.
```
	fn latest_message_id() -> BridgeMessageId {
		let nonce: u64 =
			BridgePangolinMessages::outbound_latest_generated_nonce(PANGORO_PANGOLIN_LANE).into();
		nonce_to_message_id(&PANGORO_PANGOLIN_LANE, nonce)
	}
```

**Review by HackFisher — APPROVED** [review]
LGTM, not sure whether we need to update the types.json for pangolin and pangoro.

**Comment by xiaoch05** [comment]
> LGTM, not sure whether we need to update the types.json for pangolin and pangoro.

The types.json is updated in `./node/runtime/pangolin/types.json`. @AurevoirXavier do we use the same type.json file as pangolin in pangoro?

**Comment by AurevoirXavier** [comment]
> > LGTM, not sure whether we need to update the types.json for pangolin and pangoro.
> 
> The types.json is updated in `./node/runtime/pangolin/types.json`. @AurevoirXavier do we use the same type.json file as pangolin in pangoro?

Yes, there is no need to update for pangoro.

File: node/runtime/pangolin/polkadot-compatible-types.json [hunk 1]
```diff
@@ -186,7 +186,7 @@
 			"Erc20": "TokenInfo"
 		}
 	},
-	"BridgeMessageId": "[u8; 16; BridgeMessageId]",
+	"TokenMessageId": "[u8; 16; TokenMessageId]",
```
Comment (AsceticBear) [inline node/runtime/pangolin/polkadot-compatible-types.json:189]: I am curious about this writing style, Can we change it to `"TokenMessageId": "[u8; 16]",`?

**Review by AsceticBear — APPROVED** [review]
Seems good. A question feedback left.

**Pull request merged**